Source/Models/APIClient.swift

```diff
@@ -2,19 +2,24 @@ import Foundation
 import Argo
 import Result
 
-public struct APIClient {
+public struct APIClient: Client {
   private let requestPerformer: RequestPerformer
 
+  public var queue: dispatch_queue_t? = dispatch_get_main_queue()
+
   public init(requestPerformer: RequestPerformer = NetworkRequestPerformer()) {
     self.requestPerformer = requestPerformer
   }
-}
 
-extension APIClient: Client {
   public func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseObject, SwishError> -> Void) -> NSURLSessionDataTask {
-    return requestPerformer.performRequest(request.build()) { result in
+    return requestPerformer.performRequest(request.build()) { [queue] result in
```

Comment:
Semantically, if you set the API client's queue, you've created a second client. However I agree that it's more cumbersome in practice to create a `var` and set a property than just pass the value into the initialiser. I've pushed a change for that.